Apollo/Devices/Fade.cs

```diff
@@ -326,5 +405,117 @@ class FadeInfo {
             Time.Dispose();
             base.Dispose();
         }
+
+        List<Color> GenerateLinear(Color start, Color end, int max)
+        {
+            List<Color> steps = new List<Color>();
+
+            for (int k = 0; k < max; k++)
+            {
+                steps.Add(new Color(
+                    (byte)(start.Red + (end.Red - start.Red) * k / max),
```

Comment:
but you're changing the colors here? you'll lose fidelity, and the timings will be messed up. you should apply the curve to the time axis instead, will end up looking smoother and will properly work with the smoothness settings